src/journal/catalog.c

```diff
@@ -371,21 +371,21 @@ int catalog_import_file(Hashmap *h, const char *path) {
         return 0;
 }
 
-static int64_t write_catalog(const char *database, struct strbuf *sb,
-                             CatalogItem *items, size_t n) {
-        CatalogHeader header;
+static int64_t write_catalog(
+                const char *database,
+                struct strbuf *sb,
+                CatalogItem *items,
+                size_t n) {
+
         _cleanup_fclose_ FILE *w = NULL;
-        int r;
-        _cleanup_free_ char *d, *p = NULL;
+        _cleanup_free_ char *p = NULL;
+        CatalogHeader header;
         size_t k;
+        int r;
 
-        d = dirname_malloc(database);
-        if (!d)
-                return log_oom();
-
-        r = mkdir_p(d, 0775);
+        r = mkdir_parents_label(database, 0755);
```

Comment:
This means that we need to do `mac_selinux_init()` in journalctl, no?

Reply:
I think I will drop the "label" thing again. You are right, in the current logic it wouldn't even do anything. And it's also unnecessary these days since the kernel can derive the label to use from the fname anyway these days, hence it's unnecessary too. Also, it doesn't actually fix #11903 anyway, and if we'd load the db we'd substantially make things slower, hence let's just not bother. will keep the patch in place otherwise though, i.e. that we use mkdir_parents() and the right access mode at least. Also, will add a commit that also drops the label thing from systemd-random-seed, as there too we use it but don't actually load the db first...